app/backend/src/main/java/com/example/controllers/FamilyController.java

```diff
@@ -0,0 +1,64 @@
+package com.example.controllers;
+
+import com.example.models.Family;
+import com.example.services.FamilyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("api/families")
+public class FamilyController {
+    @Autowired
+    private FamilyService familyService;
+
+    @GetMapping
+    public List<Family> getAllFamilies() {
+        return familyService.getAllFamilies();
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Family> getFamilyById(@PathVariable String id) {
+        Family family = familyService.getFamilyById(id);
+        if (family != null) {
+            return new ResponseEntity<>(family, HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+    }
```

Comment:
This does not need to be a ResponseEntity. Look into Optional type, try with that. ResponseEntity has some special use cases (which I can't name of the top of my head :D) but Optional should do just fine here. Ofc, I could also be talking out of my ass rn and be entirely wrong, we'll look into it later, it's not a big priority rn.